pyatv/core/scan.py

```diff
@@ -328,6 +329,29 @@ def _name_without_type(name: str, type_: str) -> str:
     return name[: -(len(type_) + 1)]
 
 
+def _get_valid_ipv4_address_from_service_info(
+    info: AsyncServiceInfo,
+) -> List[IPv4Address]:
+    """Return valid IPv4 addresses from service info.
+
+    Older airport devices (e.g. Airport Express) can return
+    link-local addresses in the service info after the valid
+    addresses. This function will filter out those addresses.
+    """
+    addresses = info.ip_addresses_by_version(IPVersion.V4Only)
+    if TYPE_CHECKING:
+        ipv4_addresses = cast(List[IPv4Address], addresses)
```

Comment:
Guarded the type checking here since its creating a new `List[]` typing object each time